nodeenv.py

```diff
@@ -577,17 +577,28 @@ def tarfile_open(*args, **kwargs):
         tf.close()
 
 
+def _download_node_file(node_url, n_attempt=3):
+    """Do multiple attempts to avoid incomplete data in case
+    of unstable network"""
+    while n_attempt > 0:
+        try:
+            return io.BytesIO(urlopen(node_url).read())
+        except IncompleteRead as e:
+            logger.warning(
+                'Incomplete read while reading'
+                'from {} - {}'.format(node_url, e)
+            )
+            n_attempt -= 1
+            if n_attempt == 0:
```

Comment:
This condition seems superflous together with the while loop. It would make sense to raise after the loop since you already return in `try`.